crates/cli/src/cli_args/add.rs

```diff
@@ -0,0 +1,180 @@
+use crate::State;
+use clap::Args;
+use miette::Context;
+use pacquet_package_json::DependencyGroup;
+use pacquet_package_manager::Add;
+use std::path::PathBuf;
+
+#[derive(Debug, Args)]
+pub struct AddDependencyOptions {
+    /// Install the specified packages as regular dependencies.
+    #[clap(short = 'P', long)]
+    save_prod: bool,
+    /// Install the specified packages as devDependencies.
+    #[clap(short = 'D', long)]
+    save_dev: bool,
+    /// Install the specified packages as optionalDependencies.
+    #[clap(short = 'O', long)]
+    save_optional: bool,
+    /// Using --save-peer will add one or more packages to peerDependencies and install them as dev dependencies
+    #[clap(long)]
+    save_peer: bool,
+}
+
+impl AddDependencyOptions {
+    /// Convert the `--save-*` flags to an iterator of [`DependencyGroup`]
+    /// which selects which target group to save to.
+    fn dependency_groups(&self) -> impl Iterator<Item = DependencyGroup> {
+        let &AddDependencyOptions { save_prod, save_dev, save_optional, save_peer } = self;
+        let has_prod = save_prod || (!save_dev && !save_optional && !save_peer); // no --save-* flags implies --save-prod
+        let has_dev = save_dev || (!save_prod && !save_optional && save_peer); // --save-peer with nothing else implies --save-dev
+        let has_optional = save_optional;
+        let has_peer = save_peer;
+        std::iter::empty()
+            .chain(has_prod.then_some(DependencyGroup::Default))
+            .chain(has_dev.then_some(DependencyGroup::Dev))
+            .chain(has_optional.then_some(DependencyGroup::Optional))
+            .chain(has_peer.then_some(DependencyGroup::Peer))
+    }
+}
+
+#[derive(Debug, Args)]
+pub struct AddArgs {
+    /// Name of the package
+    pub package: String,
+    /// --save-prod, --save-dev, --save-optional, --save-peer
+    #[clap(flatten)]
+    pub dependency_options: AddDependencyOptions,
+    /// Saved dependencies will be configured with an exact version rather than using
+    /// pacquet's default semver range operator.
```

Comment:
Try to minimize the amount of "pacquet" mentions in comments. Maybe we'll rename the project in the future and we don't want that to trigger unnecessary diffs. 
```suggestion
    /// the default semver range operator.
```